# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RemoteDecoderManagerChild.cpp
# Commit: c177bcba1c07
# Full Hash: c177bcba1c076d041e3470c8058bb2b4adaa1123
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - P8: Use RemoteDecoderModule to communicate with GPU and RDD. r=mattwoodrow
#   
#   Depends on D54879
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54880
# ==============================================================================

diff -r 57607f8b6d8e -r c177bcba1c07 dom/media/ipc/RemoteDecoderManagerChild.cpp
--- a/dom/media/ipc/RemoteDecoderManagerChild.cpp	Tue Oct 20 23:26:27 2020 +0000
+++ b/dom/media/ipc/RemoteDecoderManagerChild.cpp	Tue Oct 20 23:26:30 2020 +0000
@@ -303,8 +303,13 @@
 }
 
 /* static */
-void RemoteDecoderManagerChild::LaunchRDDProcessIfNeeded() {
-  if (!XRE_IsContentProcess()) {
+void RemoteDecoderManagerChild::LaunchRDDProcessIfNeeded(
+    RemoteDecodeIn aLocation) {
+  MOZ_DIAGNOSTIC_ASSERT(XRE_IsContentProcess(),
+                        "Only supported from a content process.");
+
+  if (aLocation != RemoteDecodeIn::RddProcess) {
+    // Not targeting RDD process? No need to launch RDD.
     return;
   }
 